### ISSUE TITLE
Styled input checkbox focus state is not visible on Ubuntu/Firefox

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#14354]
## Description

For checkboxes with `jp-mod-styled` class the focus state is not visible. This might be Operating System specific.

## Reproduce

1. Open search box in a notebook
2. Open filters
3. Tab to a checkbox
4. See no focus indication

On Ubuntu/Firefox

![Screenshot from 2023-04-10 15-28-33](https://user-images.githubusercontent.com/5832902/230921695-c89dbd91-8e27-4319-9a1e-3f18f0cd05b3.png)

The one on right is focused.

The styles are defined here:

https://github.com/jupyterlab/jupyterlab/blob/89154f525578e3c79d9f688be1e67605f6d2d443/packages/ui-components/style/styling.css#L45-L48

but it has no effect on styles for me in Firefox on Ubuntu.

Manually adding `outline: 3px solid red;` to `input.jp-mod-styled:focus` shows an indication properly:

![Screenshot from 2023-04-10 15-30-40](https://user-images.githubusercontent.com/5832902/230921881-ceba7417-88d0-4029-a99a-e7f7071fb8c5.png)

Of note, the styles are effective on Ubuntu/Chromium but are the indicator is rather faint:

![Screenshot from 2023-04-10 15-36-15](https://user-images.githubusercontent.com/5832902/230923107-7a814346-0064-41d4-8b7b-3298e346c57f.png)

## Expected behavior

Some kind of indicator is visible for checkboxes across browsers

## Context

- Operating System and version: Ubuntu
- Browser and version: Firefox Nightly 108
- JupyterLab version: 4.0.0b1

**Comment by JasonWeill** [comment]
Also happens on Firefox ESR 102.9 on macOS with 4.0.0b1.

**Comment by alden-ilao** [comment]
Hello! This is my first time taking a swing at open source contributions and would love some feedback if possible!

**Comment by fcollonval** [comment]
Thanks @alden-ilao for working on this. @krassowski (thanks for the review), did #15612 fix this fully?

**Comment by krassowski** [comment]
Yes, thanks for catching this!
